### PR TITLE
added component analyses check

### PR DESCRIPTION
**Pull request by yzainee**

### CODE REVIEW

File: bayesian/utils.py [hunk 1]
```diff
@@ -1223,6 +1223,19 @@ def get_recommendation_feedback_by_ecosystem(ecosystem):
...
+    "npm",
+    "maven"
+]
+
+
+def check_for_accepted_ecosystem(ecosystem):
+    """Check if the ecosystem is supported or not."""
+    if ecosystem in accepted_ecosystems:
+        return True
+    return False
```
Comment (tisnik) [inline bayesian/utils.py]: just `return ecosystem in accepted_ecosystems` is sufficient

File: tests/test_utils.py [hunk 2]
```diff
@@ -433,3 +434,12 @@ def test_get_ecosystem_from_manifest():
 
     resp = get_ecosystem_from_manifest("dependencies.txt")
     assert resp == "maven"
+
+
+def test_check_for_accepted_ecosystem():
+    """Test check_for_accepted_ecosystem function."""
+    resp = check_for_accepted_ecosystem("maven")
+    assert resp is True
```
Comment (tisnik) [inline tests/test_utils.py]: IMHO CI will fail on this line, `is` don't have to be there

File: tests/test_utils.py [hunk 3]
```diff
@@ -433,3 +434,12 @@ def test_get_ecosystem_from_manifest():
...
+
+    resp = check_for_accepted_ecosystem("abcd")
+    assert resp is False
```
Comment (tisnik) [inline tests/test_utils.py]: `assert not resp`

File: bayesian/api_v1.py [hunk 4]
```diff
@@ -293,6 +298,11 @@ def post():
             version = dependency.get('version')
             if not all([ecosystem, package, version]):
                 raise HTTPError(422, "provide the valid input.")
+            if check_for_accepted_ecosystem(ecosystem) is False:
```
Comment (tisnik) [inline bayesian/api_v1.py]: `if not check_for_accepted_ecosystem(ecosystem):`

File: bayesian/api_v1.py [hunk 5]
```diff
@@ -240,6 +240,11 @@ class ComponentAnalyses(Resource):
     def get(ecosystem, package, version):
         """Handle the GET REST API call."""
         package = urllib.parse.unquote(package)
+        if check_for_accepted_ecosystem(ecosystem) is False:
```
Comment (tisnik) [inline bayesian/api_v1.py]: `if not check_for_accepted_ecosystem(ecosystem):`

File: tests/test_utils.py (hunk 2 above)
Comment (yzainee) [reply tests/test_utils.py]: it doesnt fail. but changed anyways :)

File: tests/test_utils.py (hunk 3 above)
Comment (yzainee) [reply tests/test_utils.py]: changed

File: bayesian/api_v1.py (hunk 4 above)
Comment (yzainee) [reply bayesian/api_v1.py]: changed

File: bayesian/api_v1.py (hunk 5 above)
Comment (yzainee) [reply bayesian/api_v1.py]: changed

File: bayesian/utils.py (hunk 1 above)
Comment (yzainee) [reply bayesian/utils.py]: done

**Pull request merged**